### ISSUE TITLE
[conan-center] Prefer built in tools over recipe methods

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/conan-center-index#11949]

**Comment by jgsogo** [comment]
Although this would be useful, it is really hard from a linter perspective to check if certain function is doing certain logic. Easy as human, but almost impossible for the machine. I mean, the linter can check if there is any method in the recipe called `_is_msvc`, but it cannot know if the behavior matches the one from `tools.microsoft.is_msvc` or not. 

What would be your suggestion (in pseudo-code)?

**Comment by uilianries** [comment]
Based on what I review so far, 99% of cases are same, because we copied all from SpaceIm once.

I feel confident on suggesting that replacement, as users will need to indeed replace it, we will need to review anyway. In case something is different, is because the method does more than only checking is if msvc.

**Comment by jgsogo** [comment]
> I feel confident on suggesting that replacement, as users will need to indeed replace it, we will need to review anyway. In case something is different, is because the method does more than only checking is if msvc.

Take into account that the linter is not suggesting... **the linter is blocking** the PR from being merged.

**Comment by uilianries** [comment]
Yes, but I still prefer to see the linter blocking PRs now than I suggesting it for every review, like last few days. I feel comfortable because we will need to break it at some point, and as we are enforcing conan.tools now, it would be better do it now, instead creating a second wave of recipes updates in the future.

**Comment by jgsogo** [comment]
Ok, then go for it. If some recipe still needs it, it can rename the method to something else 😄

**Comment by prince-chrismc** [comment]
> the linter can check if there is any method in the recipe called `_is_msvc`, but it cannot know if the behavior matches the one from `tools.microsoft.is_msvc`

That's part of the problem, there's a few stale recipes that haven't gotten the `msvc` compiler 

> What would be your suggestion (in pseudo-code)?

I would be more aggressive, the tools should be correct. However the best matching I can think of

```
if has_method(conanfile, _is_msvc) and method.to_string().contains(self.settings.compiler, ==, "Visual Studio" or "msvc"):
    Report error
```
